modules/ocf_desktop/files/xsession/print-notify-listener

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+set -euo pipefail
+
+function clean_up {
+    jobs -pr | xargs pgrep -P | xargs sudo -u ocfbroker kill
+    exit 0
+}
+
+trap clean_up INT TERM EXIT
+
+pipe="$HOME/tmp/print-notify"
+icon=/opt/share/xsession/images/ocf-color-256.png
+
+[[ -p "$pipe" ]] && rm "$pipe"
+mkfifo $pipe
+
+sudo -u ocfbroker /opt/share/puppet/print-notify-handler > "$pipe" &
+
+# listener
+while true
```

Comment:
I'm assuming this doesn't busy-wait and use extra CPU by being in a loop with no pauses?

Reply:
Putting a `sleep 1` at the end of each iteration is probably a good idea.

Reply:
I mean if this doesn't use extra CPU, then you don't need to do that, but I just wanted to check if it did or not when you were testing it

Reply:
nah it was using 0%

Reply:
There is a way to keep the pipe open at all times. Before the while loop write (on a solitary line):

```bash
3< "$pipe"
```

This opens up the pipe as FD 3 in the shell script. Then in the loop below, you can replace the read statement with:

```bash
read -u 3 msg
```

Reply:
The `read` statement is blocking so this shouldn't busy-loop.

Reply:
Wow, great info. nit: before the loop it needs to be
```bash
exec 3< "$pipe" 
```
I think that's probably the best way to do this. Then we don't need to give users the perm to kill as `ocfbroker` (a one liner means no need for an extra puppet template, so +1 that), and no `xargs` nonsense on cleanup. The `cleanup` method could actually be reduced to a much more intuitive one-liner:
```bash
exec 3<&- && rm -f "$pipe"
```
This will cause the python script to die when it tries to print. That means we'll have to change the loop to a `while True` to send periodic messages (the `redis` library actually has that in their demo code, anyway, not `listen()`). This would also be the case if we were checking for the existence of a file, anyway.

Reply:
> nit: before the loop it needs to be:

Derp lol

> `rm -f "$pipe" && exec 3<&- && exit`

Well, you wouldn't need the `exit` since it's already going to exit. The `exec 3<&-` might not be required either since the FD would get closed on program exit anyway.

Reply:
> to a `while True` to send periodic messages

I don't think you would necessarily need to send periodic messages, something like

```
while True:
    message = p.get_message()
    if message:
        write it to the pipe
    poll stdout for a POLLERR event with timeout 0.1 or whatever
    if poll returns a nonempty list:
        break
```

should work.

Or if we're adventurous we could find which FD is the redis socket and directly pass that and stdout to `poll`. That's probably fragile though.